src/pages/Staking/VestsList/DelegateVest/DelegateVest.container.tsx

```diff
@@ -1,28 +1,91 @@
+import { useCallback, useEffect, useState } from 'react';
 import { useSelector } from 'react-redux';
+import { SubmitStatusDialog } from '../../../../components/TxDialog/TxDialog.component';
+import { Vesting } from '../../../../contracts/types';
+import { useContractCall } from '../../../../hooks/useContractCall';
+import {
+  accountSelector,
+  providerSelector,
+} from '../../../../store/app/app.selectors';
 
-import { selectedVestSelector } from '../../../../store/staking/staking.selectors';
+import {
+  combinedVotingPowerSelector,
+  selectedVestSelector,
+} from '../../../../store/staking/staking.selectors';
+import { getVesting } from '../../../../store/staking/staking.utils';
+import { DelegateFeeEstimator } from '../../Staking.types';
 
 import { DelegateVestComponent } from './DelegateVest.component';
+import { DelegateValues } from '../../DelegateFeeEstimator/DelegateVest.fields';
 import { DelegateVestContainerProps } from './DelegateVest.types';
 
-const mockVotingPower = '15.0000';
-
 export const DelegateVestContainer = ({
   open,
   onClose,
 }: DelegateVestContainerProps) => {
+  const account = useSelector(accountSelector);
   const selectedVestData = useSelector(selectedVestSelector);
+  const combinedVotingPower = useSelector(combinedVotingPowerSelector);
+  const provider = useSelector(providerSelector);
+  const [vesting, setVesting] = useState<Vesting | undefined>();
+
+  useEffect(() => {
+    if (!provider || !selectedVestData) {
+      throw new Error('missing data');
+    }
+    const vested = getVesting(selectedVestData.address, provider);
+
+    setVesting(vested);
+  }, [selectedVestData, provider]);
+
+  const handleDelegate = ({ delegateTo }: DelegateValues) => {
+    if (!vesting || !provider || !selectedVestData || !account) {
+      throw new Error('missing data');
+    }
+
+    return vesting.delegate(delegateTo.toLowerCase(), { from: account });
+  };
+
+  const { handleSubmit: onDelegate, ...delegateTxData } =
+    useContractCall(handleDelegate);
+
+  const estimateFee: DelegateFeeEstimator = useCallback(
+    (delegateTo) => {
+      if (!vesting || !selectedVestData) {
+        throw new Error('missing data');
+      }
+
+      return vesting.estimateGas.delegate(delegateTo);
+    },
+    [selectedVestData, vesting]
+  );
 
-  if (!selectedVestData) {
+  if (
+    !selectedVestData ||
+    !account ||
+    !combinedVotingPower ||
```

Comment:
there is no need to check both combinedVotingPower and combinedVotingPower.data since combinedVotingPower will always be an object

```suggestion
    !combinedVotingPower.data
```